lib/ruby_lsp/requests/definition.rb

```diff
@@ -67,22 +67,21 @@ def on_command(node)
         string = argument.parts.first
         return unless string.is_a?(SyntaxTree::TStringContent)
 
-        required_file = "#{string.value}.rb"
-
         case message
         when "require"
-          candidate = find_file_in_load_path(required_file)
+          candidate = @index.search_require_paths(string.value).first
```

Comment:
We currently don't have a map of `require_path => IndexablePath`. We only have the prefix tree.

So it would be a choice between searching the existing prefix tree or paying the price to keep another hash. In my opinion, we should just search the prefix tree because:

- This is a user initiated request (not a request triggered when editing the file). Which means it's not in the performance critical path
- Maintaining another hash just for this seems wasteful since it would only be used when going to a require definition
- The full prefix search is quite fast, even if not as fast as direct access. Even if it were to take a moment, VS Code displays a progress bar while it's searching for the definition. I really don't think there's any loss in terms of DX